### PR TITLE
mpvScripts.sponsorblock-minimal: init at unstable-2023-08-20

### PR DESCRIPTION
**Pull request by arthsmn**
## Description of changes

This is a simplified script to integrate mpv with Sponsorblock. This one is more limited, can only skip segments, but it also doesn't depend on python.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/buildLua.nix [hunk 1]
```diff
@@ -2,7 +2,6 @@
 , stdenvNoCC }:
 
 let
-  inherit (lib) hasPrefix hasSuffix removeSuffix;
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/buildLua.nix:5]: i'd not make unrelated changes in this PR.

File: pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  preInstall = ''
+    substituteInPlace sponsorblock_minimal.lua \
+      --replace "curl" "${lib.getExe curl}"
+  '';
+
+  meta = with lib; {
+    description = "A minimal script to skip sponsored segments of YouTube videos";
+    homepage = "https://codeberg.org/jouni/mpv_sponsorblock_minimal";
+    license = licenses.gpl3;
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix:29]: gpl3 is deprecated identifier, use gpl3Only or gpl3Plus

File: pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "mpv_sponsorblock_minimal";
+  version = "unstable-2023-08-20";
+  scriptPath = "sponsorblock_minimal.lua";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "jouni";
+    repo = "mpv_sponsorblock_minimal";
+    rev = "ca2844b8cf7674bfccd282d389a50427742251d3";
+    sha256 = "sha256-28HWZ6nOhKiE+5Ya1N3Vscd8aeH9OKS0t72e/xPfFQQ=";
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix:18]:
```suggestion
    hash = "sha256-28HWZ6nOhKiE+5Ya1N3Vscd8aeH9OKS0t72e/xPfFQQ=";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix (hunk 3 above)
Comment (arthsmn) [reply pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix:18]: Done.

File: pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix (hunk 2 above)
Comment (arthsmn) [reply pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix:29]: Done.

File: pkgs/applications/video/mpv/scripts/buildLua.nix (hunk 1 above)
Comment (arthsmn) [reply pkgs/applications/video/mpv/scripts/buildLua.nix]: Okay, removed.

**Review by onemoresuza — APPROVED** [review]
# Reviewed points

Result of `nixpkgs-review pr 273387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mpvScripts.sponsorblock-minimal</li>
  </ul>
</details>


- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [X] `meta.platforms` is set
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [X] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [X] patches that are remotely available are fetched rather than vendored

File: pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildLua
+, fetchFromGitea
+, curl
+}:
+
+# Usage: `pkgs.mpv.override { scripts = [ pkgs.mpvScripts.sponsorblock-minimal ]; }`
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix:7]: This should already be documented elsewhere, please remove

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix (hunk 4 above)
Comment (arthsmn) [reply pkgs/applications/video/mpv/scripts/sponsorblock-minimal.nix]: Done.

**Review by eclairevoyant — APPROVED** [review]
works without additional config beyond the wrapper override. lgtm
